app/models/taxon.rb

```diff
@@ -872,6 +874,21 @@ def taxon_cant_be_its_own_ancestor
       errors.add(self.name, "can't be its own ancestor")
     end
   end
+  
+  def rank_level_must_be_finer_than_parent
+    return if parent.nil?
+    
+    if (parent.rank_level.nil? || rank_level.nil?) || (parent.rank_level.to_f <= rank_level.to_f)
```

Comment:
I might get rid of the nil checks here. `rank_level.to_f` will be 0.0 if rank_level is nil, and if the rank_level or parent rank_level is nil, "rank level must be finer than parent" is a confusing error message. Better to add separate validations to check for that.